testPad/compound.testpad.test.ts

```diff
@@ -0,0 +1,329 @@
+      Assets.registerAsset(
+        "Turing",
+        "TUR",
+        10,
+        {
+          parents: 1,
+          interior: { X1: { Parachain: 2114 } },
+        },
+        537600000000
+      ),
+      sudo
+    );
+
+    testLog.getLog().info("running section: setup currency");
+
+    await signSendSuccess(
+      oakApi.api,
+      oakApi.api.tx.sudo.sudo(oakApi.addChainCurrencyData(2110, 1)),
+      alice
+    );
+
+    // send some TURs to our chain, so we can mint & send them back, otherwise XCM would fail
+    await signSendSuccess(
+      oakApi.api,
+      oakApi.api.tx.xTokens.transfer(
+        1,
+        1_000_000_000_000_000,
+        {
+          V1: {
+            parents: 1,
+            interior: {
+              X2: [
+                { Parachain: 2110 },
+                {
+                  AccountId32: {
+                    network: "Any",
+                    id: alice.keyRingPair.publicKey,
+                  },
+                },
+              ],
+            },
+          },
+        },
+        4_000_000_000
+      ),
+      alice
+    );
+  }
+
+  it("auto-compound: schedule & execute task", async () => {
+    testLog.getLog().info("running section: prepare compound extrinsic");
+
+    // @ts-ignore
+    const proxy = await oakApi.api.rpc.xcmpHandler.crossChainAccount(
+      user1.keyRingPair.address
+    );
+    const assetId = await getNextAssetId();
+    const poolId = assetId.add(BN_ONE);
+    await Sudo.batchAsSudoFinalized(
+      Assets.mintNative(user1),
+      Assets.issueToken(user1),
+      Sudo.sudoAs(user1, api.tx.proxy.addProxy(proxy, { AutoCompound: 0 }, 0)),
+      Sudo.sudoAs(
+        user1,
+        Xyk.createPool(BN_ZERO, BN_MILLION, assetId, BN_MILLION)
+      )
+    );
+
+    // it is complex to set up the chain with predefined rewards, it is enough to test without it,
+    // we add 1_000 MGR into the poolId
+    const tx = api.tx.proxy.proxy(
+      user1.keyRingPair.address,
+      undefined,
+      api.tx.xyk.provideLiquidityWithConversion(poolId, 0, BN_THOUSAND)
+    );
+    const encodedTxHex = tx.unwrap().toHex();
+    const encodedTxInfo = await tx.paymentInfo(user1.keyRingPair);
+
+    testLog.getLog().info(encodedTxHex);
+    testLog.getLog().info("running section: schedule task: " + encodedTxHex);
+
+    // @ts-ignore
+    const taskIdLog = await oakApi.api.rpc.automationTime.generateTaskId(
+      alice.keyRingPair.address,
+      "compound"
+    );
+    testLog.getLog().info(`task id: ${taskIdLog}`);
+    const taskId = "compound";
+    // const seconds = Math.trunc(new Date().getTime() / 1000);
+    // const nextHour = seconds - (seconds % HOUR) + HOUR;
+
+    const executions = [0];
+    const txCreate = oakApi.api.tx.automationTime.scheduleXcmpTask(
+      // task id
+      oakApi.api.createType("Vec<u8>", taskId),
+      // intervals for task execution, 0 == immediate execution, only with 'features=dev_queue' oak build
+      { Fixed: { executionTimes: executions } },
```

Comment:
@vanderian we are defining a new interface to call here, but the interface will be very similar to this one.
https://github.com/OAK-Foundation/oak.js/blob/main/demo/src/scheduleFixedDynamicDispatchTask.js#L67

```
const extrinsic = api.tx.automationTime.scheduleDynamicDispatchTask(providedId, { fixed: { executionTimes }}, call);
```

The `call` here is the function to be trigger on your blockchain by XCM. I picture the call to be something like `ClaimSwapDeposit`. Please check out my documentation of "TODOs for Mangata"
https://docs.google.com/document/d/1oGuHeVT9z6xwCqGwioL9P515T8wz9nsFzMmd8CWqXBE/edit#heading=h.6mqvvr281phu

Basically, in order to auto-compound, you need to create a new extrinsic to bind the Claim, Swap and Deposit in one call, and calculate how much to swap for max efficiency.

and then, in this line that call will be passed to `automationTime.scheduleXCMDynamicDispatchTask` to be inserted into Turing Network’s task registry, and we will trigger and send a message to you at the right conditions.

I’m also attaching the proposed flow chart here. Please feel free to comment on the above Google Docs.
![Turing __ Mangata XCM Flow - Page 1](https://user-images.githubusercontent.com/2616844/199288065-6bead8b4-b397-4ee6-992e-8255416dd636.png)